### ISSUE TITLE
Can't copy raw text after an HTML link is applied to it

### ISSUE THREAD

**Issue by lmattcris** [giuspen/cherrytree#1952]
**Version, Operative system**
CherryTree 0.99.45, Windows 10 Pro x64

**Describe the bug**
If I convert a text to have an HTML link I can't copy it anymore as simple text: copying it I have reference only to the link and not to visible text

**To Reproduce**
1. Write some text, select one or more words like `"my text"`
2. Click on the 'Insert/edit link' icon or press Ctrl+L
3. Insert your URI in the "To Website" field, like `"http://example.com"`, confirm; The original text is shown in link color with an underscore
4. If you select again `"my text"` and try a copy (say with Ctrl+C) in a text editor is copied only the link (`"http://example.com"`), not the text. Looks like there's no menu entry (even in the context menu) or shortcut to copy "my text" in another application

**Comment by ciranus** [comment]
>  copy (say with Ctrl+C)

Try with Ctrl+Shift+C
[Cherrytree keyboard_shortcuts](https://giuspen.com/cherrytreemanual/#_keyboard_shortcuts)

**Comment by lmattcris** [comment]
> Try with Ctrl+Shift+C [Cherrytree keyboard_shortcuts](https://giuspen.com/cherrytreemanual/#_keyboard_shortcuts)

Here I get the same result that with Ctrl+C, only the link is reported: are you successful with that sequence on Windows? 

The only workaround is by selecting a text larger than the link itself: if I select even just an additional blank space around the link then Ctrl+C and Ctrl+Shift+C both work 🤷

**Comment by ciranus** [comment]
I don't use windows.
What you can do to analyze is creating a node with one link, export this node to a file, and send it here.

**Comment by lmattcris** [comment]
I will try it on Linux tomorrow, I use it there as well

**Comment by nrhint** [comment]
I can confirm the same problem on Ubuntu 20.4.3

Selecting only "my text" then pasting either shortcut into gedit and emacs both pasted the link not the text. however, using nano I did get the correct text as well as using vi and ed. It seems to only be graphical based editors that are bothered by this as when I used emacs in the terminal I got the expected result of "my text"

**Comment by lmattcris** [comment]
I think it's a matter of how the applications handle the clipboard data, as someone can handle more complex data as well (like a picture) while others are restricted to simple text. Honestly, I've never interfaced directly with the clipboard, so I have no details.

What I can confirm on Windows is that even the simplest command-line interface (cmd.exe, TCC) do show the hyperlink and not the text.

**Comment by ciranus** [comment]
I agree it's hard to identify where CT **or** the clipboard configuration **or** pasting configuration is involved, depending on the application.

Example with Klipper clipboard (kde):
* copy "my text" from CT will show **2** different values in klipper: `my text` and `http://example.com`.

Now all depends on how I paste in Kate editor.
1) Ctrl+V -> `http://example.com`
2) Ctrl+Shift+Ins -> my text

Not an easy one, but probably something to be simplified at CT level by clarifying:
1) how to copy the text only
2) how to copy the link only
3) how to copy text+link, maybe as markdown like `[my test](http://example.com)`

**Comment by lmattcris** [comment]
I just fiddled with an application that reports in real-time Windows clipboard content. What's interesting is that CherryTree saves different data, depending on the command to copy. So Ctrl+C and Ctrl+Ins do add an HTML formatted part (for the applications that can make use of it, clearly) while Ctrl+Shift+C copies only text, as noted by @ciranus.

Anyway, the copied raw text is **always** the link, as shown in the following pictures. Then whatever is the method to paste clipboard afterward, well, only the link is available to the destination - unless I select more than the linked text, as mentioned before.
![2022-02-04 14_01_27-Window](https://user-images.githubusercontent.com/82572558/152534798-08b8e40b-d6fa-4ffe-8225-b5ceb7d7933d.png)
![2022-02-04 14_03_52-Window](https://user-images.githubusercontent.com/82572558/152534813-7b3888fb-f563-4132-bb21-156c498dbdab.png)

**Comment by ciranus** [comment]
[Editing | Notepad++ User Manual](https://npp-user-manual.org/docs/editing/)

**Paste Special** > ⇒ submenu with actions that pastes HTML or RTF, and special versions of copy/cut/paste which handle NULL and other binary characters

Note: The HTML and RTF actions paste the HTML and RTF source code from the HTML or RTF entries in the Windows Clipboard; it does not apply HTML or RTF formatting to what appears to be plain text in the Notepad++ editor window.

**Comment by lmattcris** [comment]
> Note: The HTML and RTF actions paste the HTML and RTF source code from the HTML or RTF entries in the Windows Clipboard; it does not apply HTML or RTF formatting to what appears to be plain text in the Notepad++ editor window.

Well, right... Though the problem is in the **origin**: CT _should_ fill the text with the **link text** (with Ctrl+Shift+C), not with the link URI - from my perspective, as a humble stubborn user 😅 

If I have to paste the whole HTML and extract only the link, and only in Notepad++, well... I'm better with selecting a blank after/before my text, copy, and delete it after pasting 😄

**Comment by nrhint** [comment]
Interesting. I didn't think about ctrl-v vs ctrl-fn-insert. I used a mix of
the two in my testing and didn't explicitly right click and paste to
simulate the ctrl-c when I interacted in the terminal window.

I agree that clarifying the commands for copy text only, copy link only and
copy all would be a great idea 😃
